core/base/mergeTreeDistanceMatrix/MergeTreeDistanceMatrix.h

```diff
@@ -0,0 +1,134 @@
+/// \ingroup base
+/// \class ttk::MergeTreeDistanceMatrix
+/// \author Mathieu Pont <mathieu.pont@lip6.fr>
+/// \date 2021.
+///
+/// This VTK filter uses the ttk::MergeTreeDistanceMatrix module to compute the
+/// distance matrix of a group of merge trees.
+///
+
+#ifndef _MERGETREEDISTANCEMATRIX_H
+#define _MERGETREEDISTANCEMATRIX_H
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#include <Triangulation.h>
+
+#include <FTMTree.h>
+#include <FTMTreeUtils.h>
+#include <MergeTreeBase.h>
+#include <MergeTreeDistance.h>
+
+namespace ttk {
+
+  /**
+   * The MergeTreeDistanceMatrix class provides methods to compute the edit
+   * distance between multiple merge trees and output a distance matrix.
+   */
+  class MergeTreeDistanceMatrix : virtual public Debug,
+                                  virtual public MergeTreeBase {
+
+  private:
+    bool parallelizeMatrix_ = true;
+
+  public:
+    MergeTreeDistanceMatrix() {
+      this->setDebugMsgPrefix(
+        "MergeTreeDistanceMatrix"); // inherited from Debug: prefix will be
+                                    // printed at the
+      // beginning of every msg
+    };
+    ~MergeTreeDistanceMatrix(){};
+
+    void setParallelizeMatrix(bool para) {
```

Comment:
that seems redundant with `setThreadNumber()`

Reply:
Removed in the next commit.